query-parser/src/util.rs

```diff
@@ -31,15 +29,15 @@ pub fn value_to_variable(v: &edn::Value) -> Option<Variable> {
 
 /// If the provided slice of EDN values are all variables as
 /// defined by `value_to_variable`, return a Vec of Variables.
-/// Otherwise, return an error.
-pub fn values_to_variables(vals: &[edn::Value]) -> Result<Vec<Variable>, FindParseError> {
+/// Otherwise, return the unrecognized Value.
```

Comment:
Eh?  This is very surprising.  I know you want `Either`, but a `Result` with a non-`Error`-deriving error is not Rust-y at all.  Why not return `Option<Vec<Variable>>` and make callers use `unwrap_or`?